nightfall-client/src/services/withdraw.mjs

```diff
@@ -8,73 +8,103 @@ It is agnostic to whether we are dealing with an ERC20 or ERC721 (or ERC1155).
 import config from 'config';
 import axios from 'axios';
 import gen from 'general-number';
-import { getContractInstance } from 'common-files/utils/contract.mjs';
 import logger from 'common-files/utils/logger.mjs';
 import constants from 'common-files/constants/index.mjs';
-import { Nullifier, Transaction } from '../classes/index.mjs';
-import {
-  findUsableCommitmentsMutex,
-  markNullified,
-  clearPending,
-  getSiblingInfo,
-} from './commitment-storage.mjs';
-import getProposersUrl from './peers.mjs';
-import { ZkpKeys } from './keys.mjs';
+import { waitForContract } from 'common-files/utils/contract.mjs';
+import { Transaction } from '../classes/index.mjs';
+import { computeCircuitInputs } from '../utils/computeCircuitInputs.mjs';
+import { getCommitmentsValues } from '../utils/getCommitmentValues.mjs';
+import { updateCommitments } from '../utils/updateCommitments.mjs';
 
-const { BN128_GROUP_ORDER, ZOKRATES_WORKER_HOST, PROVING_SCHEME, BACKEND, PROTOCOL, USE_STUBS } =
-  config;
+const { ZOKRATES_WORKER_HOST, PROVING_SCHEME, BACKEND, PROTOCOL, USE_STUBS } = config;
 const { SHIELD_CONTRACT_NAME } = constants;
 const { generalise } = gen;
 
-const NEXT_N_PROPOSERS = 3;
+const MAX_WITHDRAW = 5192296858534827628530496329220096n; // 2n**112n
 
 async function withdraw(withdrawParams) {
   logger.info('Creating a withdraw transaction');
   // let's extract the input items
   const { offchain = false, ...items } = withdrawParams;
   const { ercAddress, tokenId, value, recipientAddress, rootKey, fee } = generalise(items);
-  const { compressedZkpPublicKey, nullifierKey } = new ZkpKeys(rootKey);
 
-  // the first thing we need to do is to find and input commitment which
-  // will enable us to conduct our withdraw.  Let's rummage in the db...
-  const [oldCommitment] = (await findUsableCommitmentsMutex(
-    compressedZkpPublicKey,
+  const shieldContractInstance = await waitForContract(SHIELD_CONTRACT_NAME);
+
+  const maticAddress = await shieldContractInstance.methods.getMaticAddress().call();
+
+  const withdrawValue = value.bigInt > MAX_WITHDRAW ? MAX_WITHDRAW : value;
+  const commitmentsInfo = await getCommitmentsValues(
+    withdrawValue.bigInt,
+    [],
```

Comment:
These are really hard to parse with all the empty arrays, can we convert this into a named object so we don't forget what all these parameters are meant to represent?